packages/api/src/graphql/mutation.public.ts

```diff
@@ -536,6 +538,49 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
           failureURL
         })
       }
+    },
+
+    checkInvoiceStatus: {
+      type: GraphQLPublicInvoice,
+      args: {
+        id: {type: GraphQLNonNull(GraphQLID)}
+      },
+      description:
+        'This mutation will check the invoice status and update with information from the paymentProvider',
+      async resolve(root, {id}, context) {
+        const {authenticateUser, dbAdapter, paymentProviders} = context
+        const {user} = authenticateUser()
+
+        const invoices = await dbAdapter.invoice.getInvoicesByUserID(user.id)
+        const invoice = invoices.find(invoice => invoice !== null && invoice.id === id)
+        if (!invoice) throw new NotFound('Invoice', id)
+
+        const payments = await dbAdapter.payment.getPaymentsByInvoiceID(invoice.id)
+        const paymentMethods = await dbAdapter.paymentMethod.getActivePaymentMethods()
+
+        for (const payment of payments) {
+          if (!payment || !payment.intentID) continue
+
+          const paymentMethod = paymentMethods.find(
+            pm => pm && payment && pm.id === payment.paymentMethodID
+          )
+          if (!paymentMethod) continue // TODO: what happens if we don't find a paymentMethod
+
+          const paymentProvider = paymentProviders.find(
+            pp => pp && pp.id === paymentMethod.paymentProviderID
+          )
+          if (!paymentProvider) continue // TODO: what happens if we don't find a paymentProvider
+
+          const intentState = await paymentProvider.checkIntentStatus({intentID: payment.intentID})
+          await paymentProvider.updatePaymentWithIntentState({intentState, context})
+        }
+
+        // FIXME: We need to implement a way to wait for all the database
```

Comment:
I understand this is async operation from db and we expect invoice there after that? Shall we just add a loop/longpolling here?

Reply:
It's not the database operation. It's the event hook afterwards that updates other models for which we wait and i think we should have a better solution then just a loop.

Reply:
So you ok with single 100ms delay for now? If so let's go. Otherwise would go for loop/delay maybe.

Reply:
For now it should be enough. If it wouldn't be enough then calling the mutation again would return the correct result.